### PR TITLE
Allow extra options to be passed to haskell:hlint

### PR DESCRIPTION
**Pull request by KtorZ**
Highly inspired by the work done in haskell:hdevtools

### CODE REVIEW

File: ale_linters/haskell/hlint.vim [hunk 1]
```diff
@@ -26,9 +29,21 @@ function! ale_linters#haskell#hlint#Handle(buffer, lines) abort
...
+function! ale_linters#haskell#hlint#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'haskell_hlint_executable')
+endfunction
+
+function! ale_linters#haskell#hlint#GetCommand(buffer) abort
+    let l:hlintopts = ' --color=never --json'
+
+    return ale#Escape(ale_linters#haskell#hlint#GetExecutable(a:buffer))
+        \ . ' ' . ale#Var(a:buffer, 'haskell_hlint_options')
+        \ . ' ' . hlintopts . ' -'
```
Comment (w0rp) [inline ale_linters/haskell/hlint.vim]: Align line continuations with the same indentation level as the previous lines.

File: ale_linters/haskell/hlint.vim [hunk 2]
```diff
@@ -26,9 +29,21 @@ function! ale_linters#haskell#hlint#Handle(buffer, lines) abort
...
 endfunction
 
+function! ale_linters#haskell#hlint#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'haskell_hlint_executable')
+endfunction
+
+function! ale_linters#haskell#hlint#GetCommand(buffer) abort
+    let l:hlintopts = ' --color=never --json'
+
+    return ale#Escape(ale_linters#haskell#hlint#GetExecutable(a:buffer))
```
Comment (w0rp) [inline ale_linters/haskell/hlint.vim]: Try using just `'%e'` here, which will be replaced with the escaped executable name.

File: ale_linters/haskell/hlint.vim [hunk 3]
```diff
@@ -26,9 +29,21 @@ function! ale_linters#haskell#hlint#Handle(buffer, lines) abort
     return l:output
 endfunction
 
+function! ale_linters#haskell#hlint#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'haskell_hlint_executable')
+endfunction
```
Comment (w0rp) [inline ale_linters/haskell/hlint.vim]: Try using `'executable_callback': ale#VarFunc('haskell_hlint_executable'),` below instead.

**Comment by KtorZ** [comment]
Done and amended. 
Here's the patch:

<details>
<summary>diff --git a/ale_linters/haskell/hlint.vim b/ale_linters/haskell/hlint.vim</summary>
index f430901..534bef1 100644
--- a/ale_linters/haskell/hlint.vim
+++ b/ale_linters/haskell/hlint.vim

```vimscript
@@ -29,21 +29,17 @@ function! ale_linters#haskell#hlint#Handle(buffer, lines) abort
     return l:output
 endfunction
 
-function! ale_linters#haskell#hlint#GetExecutable(buffer) abort
-    return ale#Var(a:buffer, 'haskell_hlint_executable')
-endfunction
-
 function! ale_linters#haskell#hlint#GetCommand(buffer) abort
     let l:hlintopts = ' --color=never --json'
 
-    return ale#Escape(ale_linters#haskell#hlint#GetExecutable(a:buffer))
-        \ . ' ' . ale#Var(a:buffer, 'haskell_hlint_options')
-        \ . ' ' . hlintopts . ' -'
+    return '%e'
+    \      . ' ' . ale#Var(a:buffer, 'haskell_hlint_options')
+    \      . ' ' . hlintopts . ' -'
 endfunction
 
 call ale#linter#Define('haskell', {
 \   'name': 'hlint',
-\   'executable_callback': 'ale_linters#haskell#hlint#GetExecutable',
+\   'executable_callback': ale#VarFunc('haskell_hlint_executable'),
 \   'command_callback': 'ale_linters#haskell#hlint#GetCommand',
 \   'callback': 'ale_linters#haskell#hlint#Handle',
 \})
```

</details>

**Comment by KtorZ** [comment]
Done!
See 6540ec4dea09131ec241f5095197b04400e88877

**Comment by eborden** [comment]
Looks like we'll experience some conflicts
https://github.com/w0rp/ale/pull/1851

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by w0rp** [comment]
@eborden You can fix the conflicts on your end now.

**Pull request merged**